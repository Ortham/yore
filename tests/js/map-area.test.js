import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('google-map-react', () => 'GoogleMapReact');
jest.mock('react-icons/lib/fa/map-marker', () => 'FaMapMarker');

import MapArea from '../../src/gui/js/map-area';

describe('MapArea', () => {
    test('renders a map centered at (0,0) if no photo prop is set', () => {
        const mapArea = renderer.create(<MapArea />).toJSON();
        expect(mapArea).toMatchSnapshot();
    });

    test('renders a map centered at (0,0) if the photo has no location', () => {
        let photo = {};
        const mapArea = renderer.create(<MapArea photo={photo} />).toJSON();
        expect(mapArea).toMatchSnapshot();
    });

    test('renders a map centered and with a marker at the photo\'s location if it has one', () => {
        let photo = {
            location: {
                Existing: {
                    latitude: 52.0,
                    longitude: 36.2
                }
            }
        };
        const mapArea = renderer.create(<MapArea photo={photo} />).toJSON();
        expect(mapArea).toMatchSnapshot();
    });
});
