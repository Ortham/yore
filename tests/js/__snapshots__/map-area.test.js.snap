// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapArea renders a map centered and with a marker at the photo's location if it has one 1`] = `
<GoogleMapReact
  center={
    Object {
      "lat": 52,
      "lng": 36.2,
    }
  }
  zoom={5}
>
  <FaMapMarker
    lat={52}
    lng={36.2}
    style={
      Object {
        "color": "crimson",
        "height": "24px",
        "width": "24px",
      }
    }
  />
</GoogleMapReact>
`;

exports[`MapArea renders a map centered at (0,0) if no photo prop is set 1`] = `
<GoogleMapReact
  center={
    Array [
      0,
      0,
    ]
  }
  zoom={5}
/>
`;

exports[`MapArea renders a map centered at (0,0) if the photo has no location 1`] = `
<GoogleMapReact
  center={
    Array [
      0,
      0,
    ]
  }
  zoom={5}
/>
`;
